pkg/controller/container-runtime-config/helpers.go

```diff
@@ -511,15 +513,45 @@ func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec)
 	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
 		// if there is a match, return all the blocked registries except the one that matched and return an error as well
 		if reg == payloadReg {
-			blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
-			return blockedRegs, fmt.Errorf("error adding %q to blocked registries, cannot block the registry being used by the payload", payloadReg)
+			// If the payload registry doesn't have mirror rules configured for it, then don't add it to the blocked registries list
+			// TODO maybe? Add an opt-in configuration to ICSP and check if that is set as well
```

Comment:
I have quite likely forgotten; from this comment, I don’t understand what the “opt-in configuration to ICSP” is supposed to opt into.

Reply:
This was if we wanted to add a new field to either the image or ICSP CRD that is an explicit opt-in for blocking the payload registry - so kind of a two level protection, but not sure if that is really needed.

Reply:
That’s a good point, users using `blockedRegistries: foo` without an ICSP with `neverContactSource` would probably (?) end up with a policy.json entry blocking `foo` — but as long as the cluster is up, they can recover, so I don’t think we _have_ to reject that situation.

Here, in MCO, we can’t really “reject” a configuration, so I think it makes sense to be a bit conservative with validation, so that such semantics problems don’t prevent changing _other_ parts of the registry configuration, and getting to a working cluster.